core/src/test/scala/unit/kafka/admin/ReassignPartitionsClusterTest.scala

```diff
@@ -128,7 +147,8 @@ class ReassignPartitionsClusterTest extends ZooKeeperTestHarness with Logging {
     val actual = zkUtils.getPartitionAssignmentForTopics(Seq(topicName))(topicName)
     assertEquals(Seq(100, 101, 102), actual.values.flatten.toSeq.distinct.sorted)
     // The replica should be in the expected log directory on broker 102
-    assertEquals(expectedLogDir, adminClient.describeReplicaLogDir(Collections.singleton(replica)).all().get.get(replica).getCurrentReplicaLogDir)
+    assertEquals(expectedLogDir1, adminClient.describeReplicaLogDir(Collections.singleton(replica1)).all().get.get(replica1).getCurrentReplicaLogDir)
+    assertEquals(expectedLogDir2, adminClient.describeReplicaLogDir(Collections.singleton(replica2)).all().get.get(replica2).getCurrentReplicaLogDir)
```

Comment:
The replica on broker 100 is just changing the log dir. waitForReassignmentToComplete() only checks for the completion of cross broker replica movement, but not of cross log dirs movement. Perhaps we should do this in a waitUntil() loop?

Reply:
You are right. Strictly speaking we should use waitUntil() here. I have updated the patch accordingly.